pkg/services/ngalert/accesscontrol/rules.go

```diff
@@ -217,10 +210,12 @@ func (r *RuleService) AuthorizeRuleChanges(ctx context.Context, user identity.Re
 	}
 
 	for _, rule := range change.Update {
-		if err := r.HasAccessOrError(ctx, user, r.getRulesQueryEvaluator(rule.New), func() string {
-			return fmt.Sprintf("update alert rule '%s' (UID: %s)", rule.Existing.Title, rule.Existing.UID)
-		}); err != nil {
-			return err
+		if rule.AffectsQuery() {
```

Comment:
Note that there's another place that checks query permissions: `AuthorizeDatasourceAccessForRuleGroup` is used in ruler API `RouteDeleteAlertRules`. Can probably be cleaned up now.

See: https://github.com/grafana/grafana/pull/87349#discussion_r1608941044